parser.go

```diff
@@ -54,10 +55,13 @@ type Parser struct {
 
 	// structStack stores full names of the structures that were already parsed or are being parsed now
 	structStack []string
+
+	// markdownFileDir holds the path to the folder, where markdown files are stored
+	markdownFileDir string
 }
 
 // New creates a new Parser with default properties.
-func New() *Parser {
+func New(markdownFileDir string) *Parser {
```

Comment:
I think we can consider about functional options patten instead of passing variables. But it can be fixed by the new PR.

Reply:
What exactly do you mean by "functional options pattern".
Do you want me to pass this parameter as function parameter instead of injecting it into the "constructor" ?

Reply:
You can reference the following articles 😃 

- https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis
- https://gist.github.com/travisjeffery/8265ca411735f638db80e2e34bdbd3ae

Reply:
@Nerzal ping

Reply:
Okay, good articles! 
I'll fix this